app/src/main/java/org/schabi/newpipe/player/ui/MainPlayerUi.java

```diff
@@ -0,0 +1,988 @@
+            ((ViewGroup) parent).removeView(binding.getRoot());
+        }
+    }
+
+    @Override
+    public void destroy() {
+        super.destroy();
+
+        // Exit from fullscreen when user closes the player via notification
+        if (isFullscreen) {
+            toggleFullscreen();
+        }
+
+        removeViewFromParent();
+    }
+
+    @Override
+    public void destroyPlayer() {
+        super.destroyPlayer();
+
+        if (playQueueAdapter != null) {
+            playQueueAdapter.unsetSelectedListener();
+            playQueueAdapter.dispose();
+        }
+    }
+
+    @Override
+    public void smoothStopForImmediateReusing() {
+        super.smoothStopForImmediateReusing();
+        // Android TV will handle back button in case controls will be visible
+        // (one more additional unneeded click while the player is hidden)
+        hideControls(0, 0);
+        closeItemsList();
+    }
+
+    private void initVideoPlayer() {
+        // restore last resize mode
+        setResizeMode(PlayerHelper.retrieveResizeModeFromPrefs(player));
+        binding.getRoot().setLayoutParams(new FrameLayout.LayoutParams(
+                FrameLayout.LayoutParams.MATCH_PARENT, FrameLayout.LayoutParams.MATCH_PARENT));
+    }
+
+    @Override
+    protected void setupElementsVisibility() {
+        super.setupElementsVisibility();
+
+        closeItemsList();
+        showHideKodiButton();
+        binding.fullScreenButton.setVisibility(View.GONE);
+        setupScreenRotationButton();
+        binding.resizeTextView.setVisibility(View.VISIBLE);
+        binding.getRoot().findViewById(R.id.metadataView).setVisibility(View.VISIBLE);
+        binding.moreOptionsButton.setVisibility(View.VISIBLE);
+        binding.topControls.setOrientation(LinearLayout.VERTICAL);
+        binding.primaryControls.getLayoutParams().width
+                = LinearLayout.LayoutParams.MATCH_PARENT;
+        binding.secondaryControls.setVisibility(View.INVISIBLE);
+        binding.moreOptionsButton.setImageDrawable(AppCompatResources.getDrawable(context,
+                R.drawable.ic_expand_more));
+        binding.share.setVisibility(View.VISIBLE);
+        binding.openInBrowser.setVisibility(View.VISIBLE);
+        binding.switchMute.setVisibility(View.VISIBLE);
+        binding.playerCloseButton.setVisibility(isFullscreen ? View.GONE : View.VISIBLE);
+        // Top controls have a large minHeight which is allows to drag the player
+        // down in fullscreen mode (just larger area to make easy to locate by finger)
+        binding.topControls.setClickable(true);
+        binding.topControls.setFocusable(true);
+
+        if (isFullscreen) {
+            binding.titleTextView.setVisibility(View.VISIBLE);
+            binding.channelTextView.setVisibility(View.VISIBLE);
+        } else {
+            binding.titleTextView.setVisibility(View.GONE);
+            binding.channelTextView.setVisibility(View.GONE);
+        }
```

Comment:
I think this should be replaced by ``binding.titleTextView.setVisibility(isFullscreen ? View.VISIBLE : View.GONE);`` and so on:
It's shorter and easier to read in my opinion.